homeassistant/requirements.py

```diff
@@ -178,16 +178,16 @@ async def _async_process_requirements(
     kwargs: Any,
 ) -> None:
     """Install a requirement and save failures."""
-    if pkg_util.is_installed(req):
-        return
-
     if req in install_failure_history:
         _LOGGER.info(
             "Multiple attempts to install %s failed, install will be retried after next configuration check or restart",
             req,
         )
         raise RequirementsNotFound(name, [req])
 
+    if pkg_util.is_installed(req):
```

Comment:
Since this is expensive, should we cache it too?

Reply:
Or maybe we just cache it during startup, like we have a helper that maintains a cache that is automatically cleared when HA startup is finished.

Reply:
If the package is there it is quite quick..quite far down on the profile. It is only really expensive on the failure path because it has to search (more details in #47699).

<img width="421" alt="Screen Shot 2021-09-26 at 7 08 46 PM" src="https://user-images.githubusercontent.com/663432/134829043-d01942c8-ac5e-4db3-b910-8a51d4d35d34.png">

Reply:
ok that's fine then!